qiskit/tools/file_io.py

```diff
@@ -97,86 +107,135 @@ def file_datestr(folder, fileroot):
 
     return qresult, metadata
 
-def save_result_to_file(resultobj, filename, metadata=None):
-    """Save a result (qobj + result) and optional metatdata
+
+def save_result_to_file(result, filename, metadata=None):
+    """Save a result (qobj + result) and optional metadata
     to a single dictionary file.
 
     Args:
-        resultobj (Result): Result to save
-        filename (str): save path (with or without the json extension). If the file already
-        exists then numbers will be appended to the root to generate a unique filename.
-        E.g. if filename=test.json and that file exists then the file will be changed
-        to test_1.json
-        metadata (dict): Add another dictionary with custom data for the result (eg fit results)
+        result (Result): Result to save
+        filename (str): save path (with or without the json extension).
+            If the file already exists then numbers will be appended to the
+            root to generate a unique filename.
+            E.g. if filename=test.json and that file exists then the file will
+            be changed to test_1.json
+        metadata (dict): Add another dictionary with custom data for the
+            result (eg fit results)
 
     Return:
-        String: full file path
+        str: full file path
     """
-    master_dict = {}
-    master_dict['qobj'] = copy.deepcopy(resultobj._qobj)
-    master_dict['result'] = copy.deepcopy(resultobj._result)
-    if metadata is None:
-        master_dict['metadata'] = {}
-    else:
-        master_dict['metadata'] = copy.deepcopy(metadata)
-
-
-    #need to convert any ndarray variables to lists so that they can be
-    #exported to the json file
-    convert_qobj_to_json(master_dict['result'])
-
-    #if the filename has .json appended strip it off
+    master_dict = {
+        'qobj': copy.deepcopy(result._qobj.as_dict()),
+        'result': {
+            'job_id': result.get_job_id(),
+            'status': result.get_status(),
+            'result': copy.deepcopy(result._result)
+        },
+        'metadata': copy.deepcopy(metadata) or {}
+    }
+
+    # Need to convert any ndarray variables to lists so that they can be
+    # exported to the json file.
+    convert_result_to_json(master_dict['result'])
+
+    # If the filename has .json appended strip it off.
     if filename[-5:].lower() == '.json':
         filename = filename[0:-5]
 
     append_str = ''
     append_num = 0
 
-    while os.path.exists(filename+append_str+'.json'):
+    while os.path.exists(filename + append_str + '.json'):
         append_num += 1
-        append_str = '_%d'%append_num
+        append_str = '_%d' % append_num
 
-    with open(filename+append_str+'.json', 'w') as save_file:
+    with open(filename + append_str + '.json', 'w') as save_file:
         json.dump(master_dict, save_file, indent=1)
 
-    return filename+append_str+'.json'
+    return filename + append_str + '.json'
+
+
+def dict_to_qobj(qobj_dict):
```

Comment:
The same way we have Qobj.to_dict() ... could we have another classmethod like Qobj.from_dict()? What do you think?

Reply:
I toyed with the idea, but in the end decided to only include it in `file_io` at least for the moment - the rationale was two-fold:
* attempt to dis-encourage simply using that function and keep using `dicts` internally that are passed to that function (as it would kind of defeat the purpose of the PR)
* delay introducing it until we deal with the Qobj schema, as it will very likely be related and probably require a number of adjustments.

Reply:
Yeah, I'd rather have all transformations in `file_io`. Ok!